portfolio/src/main/webapp/script.js

```diff
@@ -50,3 +44,29 @@ async function getMsgUsingAsyncAwait() {
   const msg = await response.text();
   document.getElementById('msg-container').innerText = msg;
 }
+*/
+
+/**
+ * Fetches messages from the servers and adds them to the DOM.
+ */
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    // messages is an arraylist object, not a string, so we have to
+    // reference its indexes to create HTML content
+    const messageListElement = document.getElementById('msg-container');
+    messageListElement.innerHTML = '';
+    messageListElement.appendChild(
+        createListElement('Message 1: ' + messages[0]));
+    messageListElement.appendChild(
+        createListElement('Message 2: ' + messages[1]));
+    messageListElement.appendChild(
+        createListElement('Message 3: ' + messages[2]));
```

Comment:
I think it would be good to write this to not depend on the size of `messages` and turn this into a loop.